server/src/main/java/org/elasticsearch/index/store/FsDirectoryService.java

```diff
@@ -139,4 +128,44 @@ private static Directory setPreload(Directory directory, Path location, LockFact
         }
         return directory;
     }
+
+    final static class HybridDirectory extends NIOFSDirectory {
```

Comment:
Nit: the convention in our code base seems to be `static final` instead of `final static`.